### PR TITLE
Concurrent render limits

### PR DESCRIPTION
**Pull request by torkelo**
Current backend rendering service does not have any limits for current renderings. This can lead to situations where for example 50 or more rendering requests happen at the same time (via many alert rules triggering at the same time for example). This many rendering requests happening at the same time can cause severe load on the server in terms of cpu and memory. 

# Proposed solution

Introduce hard concurrent limits. 

```ini
[rendering]
concurrent_limit = 10
concurrent_limit_alerting = 5
```

These are hard limits that once breached will return an "error image". The thinking here is that we do not want to block the alert notifications due to slow rendering requests. Since these are hard limits I figured it might be good to have separate limits for the api and alerting. But not sure if this is needed. Another alternative would be to have a "soft" limit for the api (meaning the api will block and wait for free "slot") when limit reached and have alerting have a hard non blocking limit that returns the error image. 

This PR also adds error images for the other rendering error scenarios (timeout, plugin not installed, error) that are not used yet but I want these images to be aligned so figured I add them right now to make sure they look similar. 

image

### CODE REVIEW

**Comment by torkelo** [comment]
@DanCech @woodsaj since this has effects on HG some input from you would be appreciated. 

The hanging issue we noticed on 5.3-beta upgrade was related to alerting notifications reminders that caused all alert rules to send notifications on startup. This caused a lot of rendering requests that hangs the server / container. We have are fixing the issue with the issue with notification reminders in another PR. 

This PR tries to fix the issue of overloading the server with too many render requests.

**Comment by DanCech** [comment]
This seems like a good thing to have regardless.  I'm also wondering whether at upgrade time we should seed the new notification journal table with entries for the current date/time to avoid the barrage of alerts.

**Comment by bergquist** [comment]
@DanCech https://github.com/grafana/grafana/pull/13402/files should fix the initial problem.

File: pkg/services/rendering/interface.go [hunk 1]
```diff
@@ -13,15 +13,16 @@ var ErrNoRenderer = errors.New("No renderer plugin found nor is an external rend
...
+	Width           int
+	Height          int
+	Timeout         time.Duration
+	OrgId           int64
+	UserId          int64
+	OrgRole         models.RoleType
+	Path            string
+	Encoding        string
+	Timezone        string
+	ConcurrentLimit int
```
Comment (bergquist) [inline pkg/services/rendering/interface.go]: Nice! I didn't really like the fact that we sent an alerting bool before. This looks much better.

**Review by bergquist — APPROVED** [review]

**Review by DanCech — APPROVED** [review]

**Comment by marefr** [comment]
Partially fixes #13389

**Pull request merged**